### ISSUE TITLE
CUSTOM FIELDS do not support "case when"

### ISSUE THREAD

**Issue by qianqingchen** [metabase/metabase#6331]
hi

when i user metabase, i find custom fields just 'A - B'

when i use "case when A = 'a' then 'a1' end"

it's wrong

![image](https://user-images.githubusercontent.com/18594020/32427350-ef21c500-c2fa-11e7-8edd-2b2314eb030b.png)

can we support this?

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by xtagon** [comment]
I would like this too. Consider the case of enum fields - Charts shouldn't show values like `0`, `1`, `2`, they should show what categories those values indicate. Custom fields could solve this if it it supported switch statements.

**Comment by jornh** [comment]
For that requirement you can use custom remapping as documented in the Admin guide: https://metabase.com/docs/latest/administration-guide/03-metadata-editing.html#remapping-field-values

**Comment by xtagon** [comment]
@jornh Thank you, I had no idea!

**Comment by jgibbarduk** [comment]
@jornh I've tried to use the custom mapping feature using an SQL database, but I don't get the option 'custom mapping'. When I use a MongoDB, I do get the option. Do you know if this is expected?

**Comment by jornh** [comment]
@jgibbarduk no I don’t think that’s expected behavior I’m using it with a Postgres database where it’s working according to the docs. Only thing I can think of is trying doing a scan of both schema and field contents. If that doesn’t fix it for you how does the first screenshot in the doc page above differ for you? (Aka screenshot please 😄!)

**Comment by xtagon** [comment]
I got it to work with my Postgres database by setting the field type to Enum first in Metabase.

**Comment by jgibbarduk** [comment]
@jornh Ok, thanks. Just to confirm, I should be able to access the 'custom mapping' on a SQL Server database?

- Your databases: Microsoft SQL Server
- Metabase version: v0.25.2
- Metabase hosting environment: Docker
- Metabase internal database: MySQL

Screenshot of original database schema (with the field in question: 'custom_review_status'):
![image](https://user-images.githubusercontent.com/6859664/32859159-88ab541a-ca45-11e7-9dec-a4e8de946ac3.png)

Screenshot of 'custom_review_status':
![image](https://user-images.githubusercontent.com/6859664/32859228-c3367c04-ca45-11e7-8f26-cc228be8ea14.png)

I've tried every 'type' to see if any allow the "custom mapping" option to be show, but none have done so.

The DEBUG middleware log for the field show:

> 15/11/2017 20:30:1711-15 20:30:17 DEBUG middleware.log :: Result Metadata:
15/11/2017 20:30:17 ({:table_id 122,
15/11/2017 20:30:17  :special_type :type/Category,
15/11/2017 20:30:17  :name "custom_review_status",
15/11/2017 20:30:17  :source :breakout,
15/11/2017 20:30:17  :extra_info {},
15/11/2017 20:30:17  :fk_field_id 2640,
15/11/2017 20:30:17  :id 1860,
15/11/2017 20:30:17  :visibility_type :normal,
15/11/2017 20:30:17  :display_name "Custom Review Status",
15/11/2017 20:30:17  :base_type :type/Integer}

Any help would be great. Currently I can only use 1,2,3 as labels on my charts, which is making me sad :(

**Comment by jornh** [comment]
@jgibbarduk I just succeeded connecting Metabase to a SQL Server, see https://github.com/metabase/metabase/issues/6056#issuecomment-344762412. 

So I went ahead and tried getting 'custom mapping' to work there as I have previously done with Postgres. I see the same behavior as you - with only "Use original value" showing up - so it's not working for me either with SQL Server 😞. 

I think you should report the above as a separate issue to not clutter this issue up further as it's essentially a different feature request.

**Comment by matthelm** [comment]
It’d be fantastic to see this added. Here’s an example usecase which, as far as I can tell, currently requires SQL mode.

```sql
SELECT 
  DATE_TRUNC(‘month’, created_at) AS month
  (1.0 * SUM(CASE WHEN returned THEN 1 END) / COUNT(1)) AS return_rate
FROM orders
```

**Comment by MarkRx** [comment]
According to the documentation custom mapping only works on numeric fields (why?). It doesn't fit all cases. In my case I want to show a different value when a string field is null.

**Comment by sbelak** [comment]
Implemented in #11868
